plugins/catalog-react/src/components/UserListPicker/useStarredEntitiesCount.test.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { CatalogApi } from '@backstage/catalog-client';
+import { EntityListProvider, useStarredEntities } from '../../hooks';
+import { catalogApiRef } from '../../api';
+import { ApiRef } from '@backstage/core-plugin-api';
+import { MemoryRouter } from 'react-router-dom';
+import { useStarredEntitiesCount } from './useStarredEntitiesCount';
+import { renderHook } from '@testing-library/react-hooks';
+
+const mockQueryEntities: jest.MockedFn<CatalogApi['queryEntities']> = jest.fn();
+const mockCatalogApi: jest.Mocked<Partial<CatalogApi>> = {
+  queryEntities: mockQueryEntities,
+};
```

Comment:
These mocks are kind of complex and recurring. I wonder if it'd actually be nice to just copy over a lightly tweaked version of [`simpleMock`](https://github.com/backstage/backstage/blob/116fa16ffc9dbcd8eca42fb4880cbb76f80103e6/packages/backend-test-utils/src/next/services/mockServices.ts#L74) and making a mockCatalogClient instance with it. Then you can access its methods directly in jest.Mocked form and make assertions on them.

For a small example of usage: https://github.com/backstage/backstage/blob/116fa16ffc9dbcd8eca42fb4880cbb76f80103e6/plugins/catalog-backend-module-incremental-ingestion/src/module/catalogModuleIncrementalIngestionEntityProvider.test.ts#L40

Reply:
yeah I agree that they are more verbose, but on the other side I feel like if someone jumps on a test they've never seen, it helps people understanding what is going on in the test, as the mocked functions are immediately listed at the top, instead of being hidden and mocked by some general mocking utility 🤷‍♂️